logic/common/file_utils.py

```diff
@@ -130,4 +131,30 @@ def StripFilename(file_path):
     # Split the filename and the extension
     filename_without_extension, _ = os.path.splitext(filename_with_extension)
 
-    return filename_without_extension
\ No newline at end of file
+    return filename_without_extension
+
+
+
+
+
+def GetFile(filename):
+    '''Create new directory if it does not exist'''
+    # TODO Also read new file and return its content?
+
+    # Can extract the folder's path even if provided with a file instead
+    if filename.endswith("/"):
+        folder_path = filename
+    else:
+        folder_path = "/".join( filename.split("/")[0:-1] )
```

Comment:
The variable folder_path looks like it only exists within the scope of this if-else statement.

In other languages, this would throw a compiler error. Python allows for this technically, but it's not good coding practice. It's better to declare variables in a way that makes their scope clear to anyone reading the code.

Reply:
I usually declare something like `string folder_path;` before the if-statement, I blame it on Python!